linkerd/app/src/env.rs

```diff
@@ -500,6 +507,18 @@ pub fn parse_config<S: Strings>(strings: &S) -> Result<super::Config, EnvError>
     })
 }
 
+fn convert_string_to_map(labels: String) -> HashMap<String, String> {
+    let mut labels_map: HashMap<String, String> = HashMap::new();
+    for line in labels.lines() {
+        let label = line.split("=").collect::<Vec<&str>>();
+        labels_map.insert(
+            label[0].to_string(),
+            label[1][1..label[1].len() - 1].to_string(),
```

Comment:
If we `collect` the iterator returned by `split` to a `Vec` and then index it, we might panic if an index is out of bounds (e.g. if the line didn't contain a `=` to split on, and the iterator only had one item in it). 

Instead, I think we should just skip any lines that are invalid. We could do this by just calling `next` on the iterator twice. Since [`Iterator::next`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#tymethod.next) returns an `Option`, we'll know if the expected number of elements is missing. Maybe something like
```rust
        let mut label = line.split("=");
        if let (Some(key), Some(val)) = (labels.next(), labels.next()) {
            // the line contained both a `key` and a `value`, trim them and insert into the map...
         }
         // otherwise, the line was malformed, continue iterating...
```